docs/source/distributed.rst

```diff
@@ -361,6 +361,9 @@ is guaranteed to support two methods:
 * ``wait()`` - in the case of CPU collectives, will block the process until the operation is completed. In the case
   of CUDA collectives, will block until the operation has been successfully enqueued onto a CUDA stream and the
   output can be utilized on the default stream without further synchronization.
+* ``get_future()`` - returns ``torch._C.Future`` object. Supported for NCCL, also supported for most operations on GLOO
+  and MPI, except for peer to peer operations.
+  Note: as we continue adopting Futures and merging APIs, ``get_future()`` call might become redundant.
```

Comment:
Is there any relevant GH issue to describe the future merging work?

Reply:
not yet, I should file a small RFC on which functions should be exposed in the resulting merge